aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,312 @@
+
+Dependency Injection
+^^^^^^^^^^^^^^^^^^^^
+Filters are not directly returned from :doc:`dependency injection </fundamentals/dependency-injection>` (DI). Filters that are also attributes, and which are added directly to controller classes or action methods cannot have constructor dependencies provided by DI - their constructor properties must be specified when they are declared. This is a limitation of how attributes work. However, if your filters have dependencies you need to access from DI, there are several supported approaches. You can apply your filter using either the `ServiceFilter <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/ServiceFilterAttribute/index.html>`_ or `TypeFilter <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TypeFilterAttribute/index.html>`_ attribute, or you can implement IFilterFactory on your attribute, or you can access the services you require from the context parameter provided to your filter's On\ *Stage* methods.
+
+For an example of this last technique, consider the following method, which needs to access a service that has been registered with DI. The service is accessible from the context parameter:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/OrderLoggingActionFilterAttribute.cs
+  :language: c#
+  :emphasize-lines: 3-4
+  :lines: 10-20
+  :dedent: 8
+
+Cancellation and Short Circuiting
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+You can short-circuit the filter pipeline at any point by setting the ``Result`` property on the context parameter provided to the filter method. For instance, the following ``ShortCircuitingResourceFilter`` will prevent any other filters from running later in the pipeline, including any action filters.
+
+.. _short-circuiting-resource-filter:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/ShortCircuitingResourceFilterAttribute.cs
+  :language: c#
+  :emphasize-lines: 12-15
+
+When this filter is applied to an action that also would have an action filter applied (later in the pipeline), such as the ``SomeResource`` action below, the action filter is not executed (in this case the ``AddHeader`` action filter is applied at the class level, but the behavior would be the same if it were applied to the method).
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/SampleController.cs
+  :language: c#
+  :emphasize-lines: 1,4
+  :lines: 6-8, 14-19
+  :dedent: 4 
+
+Configuring Filters
+-------------------
+Filters can be *scoped* at three different levels. You can add a particular filter to a particular action as an attribute. You can add a filter to all actions within a controller by applying an attribute at the controller level. Or you can register a filter globally, to be run with every MVC action.
+
+Filters are added globally in ``Startup``, when configuring MVC:
```

Comment:
one line too many. You don't want to show 

`// This method gets called by the runtime. Use this method to configure the HTTP request pipeline.`